compiler/damlc/daml-stdlib-src/DA/Text.daml

```diff
@@ -53,6 +53,25 @@ module DA.Text
   , DA.Text.sha256
   , DA.Text.toCodePoints
   , DA.Text.fromCodePoints
+
+#ifdef DAML_UNSTABLE
```

Comment:
This is a very bad idea. The stdlib should never be different (in terms of types and existence of functions) depending on the _target language_. 
Also, having functions with different names makes it very cumbersome to compare anything.
To achieve the goal of comparing performance impacts for these primitives, I suggest either leaving this on a branch (if we are unsure about whether anything works) or branching the _existing_  `DA.Text` interface, like so
```
splitOn : Text -> Text -> [Text]
#ifdef DAML_UNSTABLE
splitOn = primitive  @"BETextSplitOn"
#else
splitOn "" t' = [t']
splitOn _ "" = [""]
splitOn ... current master
#endif
```
Using separate pragmas instead of a single `DAML_UNSTABLE` you can then implement a "switchboard" for test setups (testing on a branch) to measure the individual primitives or combinations.
Obviously more startup cost, but the only thing that makes sense for measurements (nobody will sit down inserting `unstable` prefixes at all the call sites for such measurements)

Reply:
> This is a very bad idea. The stdlib should never be different (in terms of types and existence of functions) depending on the _target language_.

This is not possible in general. E.g., consider the addition of the new map type. There is no way to have this for older LF versions.

Branching the existing DA.Text interface means that we have to recompile the SDK for every experiment we want to run which seems much more cumbersome than running a search and replace to add `unstable` prefixes.

Reply:
I agree with @cocreature here. Recompiling the compiler and the SDK for each experiment isn't really a good idea. It's not only extremely slow but also has the potential to introduce unrelated compilation artifacts into the benchmarks. By exposing all of these unstable functions for the same target (1.dev), we're always testing the same SDK, compiler, stdlib packages, and the only question is "do these primitives result in a significant improvement"? (That said, we don't currently have a way to make any conclusions about statistical significance in DAML. All we can do is eyeball it . . .)

The point of `DAML_UNSTABLE` is that it's only available for those working on 1.dev and it's 100% not a stable interface, so it's perfect for this kind of experimentation. We already have different parts of the stdlib that depending on the LF target (see Numeric, GenMap, AnyTemplate, etc features), because otherwise we would have to drop support for older LF versions.

Reply:
What about in a `Data.Text.Unstable` module?